clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsClusterProvider.java

```diff
@@ -36,4 +46,50 @@ public EcsClusterProvider(Cache cacheView) {
   public Collection<EcsCluster> getAllEcsClusters() {
     return ecsClusterCacheClient.getAll();
   }
+
+  public Collection<Cluster> getAllEcsClustersDescription(String account, String region) {
+    List<String> clusters = new ArrayList<>();
+    List<Cluster> response = new ArrayList<>();
+    int clusterCount = 0;
+
+    Collection<EcsCluster> ecsClusters = ecsClusterCacheClient.getAll();
+
+    if (ecsClusters.size() > 0) {
```

Comment:
You also want to check that ecsClusters is not null (this could throw an npe as is)

Reply:
Changes have been made.